AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs

```diff
@@ -6,49 +6,96 @@
     using System.Threading.Tasks;
 
     /// <summary>
-    /// Class to contain the one cache for all Gauges 
+    /// Class to contain the one cache for all Gauges.
     /// </summary>
     internal class CacheHelper
     {
+        private static readonly CacheHelper instance = new CacheHelper();
+
+        static CacheHelper() { }
+
+        private CacheHelper() { }
+
+        /// <summary>
+        /// Return the only instance of CacheHelper.
+        /// </summary>
+        public static CacheHelper Instance
+        {
+            get
+            {
+                return instance;
+            }
+        }
+
+        /// <summary>
+        /// Searchs for the value of a given performance counter in a json.
+        /// </summary>
+        /// <param name="performanceCounterName"> The name of the performance counter.</param>
+        /// <param name="json"> String containing the json.</param>
+        /// <returns> Value of the performance counter.</returns>
+        public int PerformanceCounterValue(string performanceCounterName, string json)
+        {
+            string jsonSubstring = json.Substring(json.IndexOf(performanceCounterName), json.Length - json.IndexOf(performanceCounterName));
+
+            int startingIdx = jsonSubstring.IndexOf(" ");
+            int endingIdx = jsonSubstring.IndexOf(",");
+
+            string value = jsonSubstring.Substring(startingIdx + 1, endingIdx - startingIdx - 1);
+
+            return Convert.ToInt32(value);
+        }
+
         /// <summary>
         /// Checks if a key is in the cache and if not
         /// Retrieves raw counter data from Environment Variables
         /// Cleans raw JSON for only requested counter
         /// Creates value for caching
         /// </summary>
-        /// <param name="name">cache key and name of the counter to be selected from JSON</param>
+        /// <param name="name">Cache key and name of the counter to be selected from JSON</param>
         /// <returns>value from cache</returns>
-        public static int GetCountervalue(string name)
+        public int GetCounterValue(string name)
         {
             const string jsonKey = "json";
```

Comment:
Is there only ever one value stored in the cache?  Shouldn't the jsonKey be something like "jsonKey" + name?